joss-paper/paper.md

```diff
@@ -0,0 +1,134 @@
+---
+title: 'kokkos-fft: A shared-memory FFT for the Kokkos ecosystem'
+tags:
+  - C++
+  - FFT
+  - High performance computing
+  - Performance portability
+authors:
+  - name: Yuuichi Asahi
+    orcid: 0000-0002-9997-1274
+    equal-contrib: true
+    affiliation: "1" # (Multiple affiliations must be quoted)
+  - name: Thomas Padioleau
+    orcid: 0000-0001-5496-0013
+    equal-contrib: true
+    affiliation: "1" # (Multiple affiliations must be quoted)
+  - name: Paul Zehner
+    orcid: 0000-0002-4811-0079
+    equal-contrib: true
+    affiliation: "1" # (Multiple affiliations must be quoted)
+  - name: Julien Bigot
+    orcid: 0000-0002-0015-4304
+    equal-contrib: true
+    affiliation: "1" # (Multiple affiliations must be quoted)
+affiliations:
+ - name: Université Paris-Saclay, UVSQ, CNRS, CEA, Maison de la Simulation, 91191, Gif-sur-Yvette, France
+   index: 1
+   ror: 00hx57361
+
+date: 18 April 2025
+bibliography: paper.bib
+
+# Optional fields if submitting to a AAS journal too, see this blog post:
+# https://blog.joss.theoj.org/2018/12/a-new-collaboration-with-aas-publishing
+aas-doi: 10.3847/xxxxx <- update this with the DOI from AAS once you know it.
+aas-journal: Astrophysical Journal <- The name of the AAS journal.
+---
+
+# Summary
+
+kokkos-fft implements local (non-distributed) interfaces between Kokkos and de facto standard FFT libraries, including FFTW, cufft, hipfft (rocfft), and oneMKL. The key concept of kokkos-fft is "As easy as numpy, as fast as vendor libraries". This library has simple interface like numpy.fft with in-place and out-of-place transforms, while keeping the performance portability. An exercise to solve the 2D Hasegawa-Wakatani turbulence with the Fourier spectral method demonstrates that we can achieve a signficiant speed up against python version without increasing the lines of code (LOC) drastically. This library allows the user to perform FFTs efficiently and simply in the kokkos eco system.
```

Comment:
```suggestion
kokkos-fft implements local (non-distributed) interfaces between Kokkos and de facto standard FFT libraries, including FFTW, cuFFT, rocFFT, and oneMKL. The key concept of kokkos-fft is "As easy as numpy, as fast as vendor libraries". This library has simple interface like numpy.fft with in-place and out-of-place transforms, while keeping the performance portability. An exercise to solve the 2D Hasegawa-Wakatani turbulence with the Fourier spectral method demonstrates that we can achieve a signficiant speed up against python version without increasing the lines of code (LOC) drastically. This library allows the user to perform FFTs efficiently and simply in the kokkos eco system.
```
Spelling of the FFT libs. Also I would just say rocFFT and omit hipFFT for simplicity.